chore(rest): adds `uses` counter to Connector

zregvart opened the pull request:
This adds a uses counter to Connector that counts the number of integrations that make use of a particular connector.

This in turn means that endpoints for getting and listing connectors need to load all integrations

Fixes #666

zregvart commented:
Seems that I've miss merged this

jimmidyson commented:
yeah some confusing stuff in there - let me know when it's ready to review.

One quick note - I do have @rhuss's voice in my head now about obfuscated functional code... might want to change some of that to be more imperative.

zregvart commented:
@jimmidyson I fixed an error prone warning and rebased so it should be OK now. Not sure what functional code can be refactored as imperative -- our current field types use `Optional` a lot and it's difficult not to write in functional style.
